components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -1931,31 +1931,53 @@ private void processPatchRequest(List<Patch> patchRequest, IdentityProvider idpT
             // We support only 'REPLACE' patch operation.
             if (operation == Patch.OperationEnum.REPLACE) {
                 String value = patch.getValue();
-                switch (path) {
-                    case Constants.NAME_PATH:
-                        idpToUpdate.setIdentityProviderName(value);
-                        break;
-                    case Constants.DESCRIPTION_PATH:
-                        idpToUpdate.setIdentityProviderDescription(value);
-                        break;
-                    case Constants.IMAGE_PATH:
-                        idpToUpdate.setImageUrl(value);
-                        break;
-                    case Constants.IS_PRIMARY_PATH:
-                        idpToUpdate.setPrimary(Boolean.parseBoolean(value));
-                        break;
-                    case Constants.IS_ENABLED_PATH:
-                        idpToUpdate.setEnable(Boolean.parseBoolean(value));
-                        break;
-                    case Constants.IS_FEDERATION_HUB_PATH:
-                        idpToUpdate.setFederationHub(Boolean.parseBoolean(value));
-                        break;
-                    case Constants.HOME_REALM_PATH:
-                        idpToUpdate.setHomeRealmId(value);
-                        break;
-                    default:
-                        throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
-                                .ERROR_CODE_INVALID_INPUT, null);
+                if (path.contains(Constants.CERTIFICATE)) {
```

Comment:
if we do contains here, it will allow invalid paths as well. Its better to define a regex for this and do a regex match.
e.g. you can use a regex similar to 'certificate\/certificates\/\d+'

Reply:
Fixed https://github.com/wso2/identity-api-server/pull/114/commits/d8c03ce82b78b02cebb6f87be8ce9e77c1973a80